gto/git_utils.py

```diff
@@ -14,7 +14,7 @@
 def git_clone_remote_repo(f: Callable):
```

Comment:
@aguschin I was thinking to rename this as `git_clone_on_remote_repo`. What do you think?

Reply:
Doesn't matter much for me TBH, anything will work. Maybe something like `git_clone_if_needed`?

Reply:
If I would see a decorator named `git_clone_if_needed` I would wonder "when could it be needed? Perhaps if a repo is remote? Perhaps if one does not want to mess the current repo and prefer to do a dry-run? Perhaps... why didn't they specify it? It must be something complex!"

I think it is better to be specific whenever possible. `git_clone_if_repo_is_remote` could also work, but it makes the test names really hard to read (e.g. `test_git_clone_remote_repo_if_repo_is_a_meaningless_string_then_leave_it_unchanged` would become `test_git_clone_if_repo_is_remote_if_repo_is_a_meaningless_string_then_leave_it_unchanged`).
This is why I was looking for an alternative to a `if` preposition and I came up with the idea of `on`.

Reply:
Got it. To me both `test_git_clone_remote_repo_if_repo_is_a_meaningless_string_then_leave_it_unchanged` and `become test_git_clone_if_repo_is_remote_if_repo_is_a_meaningless_string_then_leave_it_unchanged` look understandable though very long, but I assume it's normal for tests)

Interesting idea about dry-run, btw. Didn't consider this before.

As the other option, I think even `clone_remote_repo` will be fine. There is no other clone in this context except for `git clone`, so it's shouldn't be ambiguous.

Reply:
Not merging this, in case you want to rename things. Btw, are you able to merge this PR?

Reply:
I eventually think renaming as `clone_on_remote_repo` made a lot of sense, also for consistency with the new `auto_push_on_remote_repo`.
https://github.com/iterative/gto/pull/286/commits/4dd46fe3bde14fabed366426915fe1fc8df52ef0

No, I cannot merge, I can only close it.